### DISCUSSION TITLE
Can I change the format of the Rich logging handler? (to work around broken terminal width detection)

### DISCUSSION THREAD

**Discussion by Hubro** [Textualize/rich#2778] [Q&A]
I'm in love with the Rich logging handler, but unfortunately I'm also addicted to `hivemind` and `overmind`, two awesome process managers for development.

When I run a process using the Rich logging handler in `hivemind` or `overmind`, Rich fails to detect the proper terminal width:

![image](https://user-images.githubusercontent.com/597206/214809957-828e13f5-81a2-4912-8fe5-e2c842307b81.png)

It assumes 100 columns and wraps log messages unnecessarily.

To work around this, is there any way I can reconfigure the rich handler to put the filename/linenumber on a different column and pretend it has infinite columns to work with? (i.e. never wrap)

**Comment by willmcgugan** [comment]
You probably have a `COLUMNS` env var set.

**Reply by Hubro** [reply]
I have manually set a value to `$((COLUMNS-20))` to work around this issue. This works with `hivemind`, but not `overmind`. It's not perfect though, since the value doesn't update when I resize the terminal.

If I unset it (`unset COLUMNS`), Rich seems to default to 80 rather than detect the actual terminal width, so something is definitely wrong. I printed the value of `os.environ["COLUMNS"]` in Python and it's empty.
